### DISCUSSION TITLE
Bad request: 401

### DISCUSSION THREAD

**Discussion by kioannou97** [offa/influxdb-cxx#146] [Q&A] [answered]
Hello everyone, I am quite new in C++ development, and this is the first time I ever work with InfluxDB. I 've been running the following code and I keep getting a "Bad request: 401 error". The influxdb version that I use is v2.5.1.

main.cpp
```
#include <iostream>
#include <InfluxDBFactory.h>

int main()
{
	auto db = influxdb::InfluxDBFactory::Get("http://localhost:8086?db=test");
	const auto response = db->execute("SHOW DATABASES");
	return 0;
}

```
CMakeLists.txt
```
cmake_minimum_required(VERSION 3.12.0)
set(CMAKE_CXX_STANDARD 17)
project(cxx-influxdb-example)
find_package(InfluxDB)
add_executable(example main.cpp)
target_link_libraries(example PRIVATE `InfluxData::InfluxDB)
```

Can anyone help me understand why I am getting this error? Any input is greatly appreciated!

**Comment by offa** [comment, marked as answer]
401 means unauthorized, you'll need credentials for authentication.

**Reply by kioannou97** [reply]
Thank you for your quick response!

I tried to use my credentials in the following way: 

```
#include <iostream>
#include <InfluxDBFactory.h>

int main()
{
	auto db = influxdb::InfluxDBFactory::Get("http://username:password@localhost:8086?db=test");
	const auto response = db->execute("SHOW DATABASES");
	return 0;
}
```

However, I am still getting the same error.

**Reply by offa** [reply]
Can you have a look at the links listed [here](https://github.com/offa/influxdb-cxx/issues/109#issuecomment-1022332344)? Maybe there's a compatibility issue regarding token or password.

Also can you use your credentials eg. using the CLI?

**Reply by kioannou97** [reply]
I can use the credentials through the CLI just fine. I had a  look at the links that you posted and tried some other methods but didn't  work either. 

So I just downgraded to InfuxDB 1.8 and the library worked fine! I hope that in the future there will be support for  InfluxDB 2.0+.

Thank you so much for your help!

**Reply by offa** [reply]
Can we mark as solved here?

**Reply by kioannou97** [reply]
Yes :)
